packages/astro/src/content/runtime.ts

```diff
@@ -153,6 +163,21 @@ export function createGetEntryBySlug({
 
 export function createGetDataEntryById({ getEntryImport }: { getEntryImport: GetEntryImport }) {
 	return async function getDataEntryById(collection: string, id: string) {
+		const store = await globalDataStore.get();
+
+		if (store.hasCollection(collection)) {
+			const data = store.get(collection, id);
+			if (!data) {
+				throw new Error(`Entry ${collection} → ${id} was not found.`);
```

Comment:
Should we consider an `AstroError` for the future, or was this intentional?

Reply:
I've not created any new AstroErrors yet. For this one I just used the same approach as the one below, but it would make sense to use an AstroError for both.